raiden_libs/network/matrix/client.py

```diff
@@ -3,16 +3,34 @@
 from urllib.parse import quote
 
 import gevent
+from gevent.lock import Semaphore
+from matrix_client.api import MatrixHttpApi
 from matrix_client.client import CACHE, MatrixClient
 from matrix_client.errors import MatrixRequestError
 from matrix_client.user import User
+from requests.adapters import DEFAULT_POOLSIZE
 
 from .room import Room
 
 
 logger = logging.getLogger(__name__)
 
 
+class GMatrixHttpApi(MatrixHttpApi):
+    """
+    A wrapper around MatrixHttpApi to limit the number
+    of concurrent requests we make to the number of connections
+    available to us in requests.Session connection pool size.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.lock = Semaphore(DEFAULT_POOLSIZE)
+
+    def _send(self, *args, **kwargs):
+        with self.lock:
```

Comment:
I think we should prioritise calls to the `sync` endpoint because without those neither messages nor presence updates will arrive.

Reply:
The `/sync` endpoint should be the only one taking longer than a few ms. I think there's no need to prioritize, as its time will come soon enough. This lock should be locking only on peak times like startup time, so I think there's no need to make the algorithm here more complex and specific.

Reply:
Ok, fair enough, let's see how it works.

Reply:
This is the wrong fix IMO, the warnings was innocuous and this fix it limiting artificially limiting throughput.

If we just want to disable the warning, the `logging.configure` should be used and not introduce a lock.
If we actually want to limit the number of concurrent requests, then we should use urllib3's built-in feature instead of rolling our own [ref](https://urllib3.readthedocs.io/en/latest/reference/index.html#urllib3.connectionpool.HTTPConnectionPool):

Note: I'm bringing this up because the it's was unclear what this lock was doing [on another PR](https://github.com/raiden-network/raiden-libs/pull/117#discussion_r207748408)

> block – If set to True, no more than maxsize connections will be used at a time. When no free connections are available, the call will block until a connection has been released. This is a useful side effect for particular multithreaded situations where one does not want to use more than maxsize connections per host to prevent flooding.

Reply:
@hackaugusto yes, we want to limit the number of parallel connections. They usually aren't the bottleneck, as was just flooding the server on burst times like startup. I wasn't aware of this parameters. There's a [corresponding one in HTTPAdapter construtor](http://docs.python-requests.org/en/master/api/#requests.adapters.HTTPAdapter), so I'll switch to it in the other PR